IPv4 to IPv6 address mapping is disabled on OpenBSD, so do not
set DualMode in mono either.

Index: mcs/class/System/System.Net.Sockets/Socket.cs
--- mcs/class/System/System.Net.Sockets/Socket.cs.orig
+++ mcs/class/System/System.Net.Sockets/Socket.cs
@@ -148,7 +148,7 @@ namespace System.Net.Sockets
 					this.DontFragment = false;
 					if (protocolType == ProtocolType.Tcp)
 						this.NoDelay = false;
-				} else if (addressFamily == AddressFamily.InterNetworkV6) {
+				} else if (addressFamily == AddressFamily.InterNetworkV6 && !Platform.IsOpenBSD) {
 					this.DualMode = true;
 				}
 
